File: week2/hanoi.py
import sys

def hanoi(n:int, a=1, b=2, c=3) -> int:
    if n == 0:
        return
    elif n == 1:
        print(f"Move {a} to {c}")
    else:
        hanoi(n-1, a, c, b)
        print(f"Move {a} to {c}")
        hanoi(n-1, b, a, c)

if __name__ == '__main__':
    if len(sys.argv) == 2:
        n = int(sys.argv[1])
        hanoi(n)

File: week02/hanoi.cpp
#include <iostream>

int hanoi(int n, int a=1, int b=2, int c=3) {
    if (n == 0) {
        return 0;
    }
    else if (n == 1) {
        std::cout << "Move " << a << " to " << c << std::endl;
        return 0;
    }
    else {
        hanoi(n-1, a, c, b);
        std::cout << "Move " << a << " to " << c << std::endl;
        hanoi(n-1, b, a, c);
        return 0;
    }
}

int main(int argc, char* argv[]) {
    if (argc == 2) {
        int n = atoi(argv[1]);
        hanoi(n);
    }
}